### PR TITLE
test: improve coverage of pkg/quay

### PR DESCRIPTION
**Pull request by tnevrlka**
pkg/quay currently has some completely uncovered functions, making the coverage 33 %
This PR adds tests in order to increase coverage

### CODE REVIEW

File: pkg/quay/quay_test.go [hunk 1]
```diff
@@ -425,3 +425,180 @@ func TestQuayClient_DeleteTag(t *testing.T) {
...
+
+	// Get existing robot account
+	gock.New("https://quay.io/api/v1").
+		MatchHeader("Content-Type", "application/json").
+		MatchHeader("Authorization", "Bearer authtoken").
+		Get(fmt.Sprintf("organization/%s/robots/%s", org, robotName)).
+		Reply(200).
+		JSON(robot)
+
+	if received_robot, err := quayClient.GetRobotAccount(org, robotName); err != nil {
```
Comment (tisutisu) [inline pkg/quay/quay_test.go:555]: nit: Is it better to avoid snake case and stick to camel case `received_robot`

File: pkg/quay/quay_test.go [hunk 2]
```diff
@@ -425,3 +425,180 @@ func TestQuayClient_DeleteTag(t *testing.T) {
...
+	} else if *received_robot != robot {
+		t.Error("received robot is not the same as original")
+	}
+
+	// Try to get non-existing robot account
+	gock.New("https://quay.io/api/v1").
+		MatchHeader("Content-Type", "application/json").
+		MatchHeader("Authorization", "Bearer authtoken").
+		Get(fmt.Sprintf("organization/%s/robots/%s", org, robotName)).
+		Reply(404)
```
Comment (tkdchen) [inline pkg/quay/quay_test.go:566]: I'm not familiar with gock. If `.JSON(...)` is not called, what fake value will be used by gock? Is it same as the call `.JSON(map[string]string{})`?

**Review by tkdchen — COMMENTED** [review]
Nice! Can you please cover more code path, for example the error when server does not respond data in an encoded JSON string ?

Another suggestion, can you reorganize the test structure as https://github.com/redhat-appstudio/build-service/blob/main/controllers/component_build_controller_unit_test.go#L42? So that in a test, we can clearly define fake input (simulate Quay.io response) and the expected return value for each test case.

**Comment by tnevrlka** [comment]
Is it possible that `GetRobotAccount` is behaving unexpectedly? Could someone please double check?
I believe it does not return error when Unauthorized

File: pkg/quay/quay_test.go (hunk 1 above)
Comment (tnevrlka) [reply pkg/quay/quay_test.go]: Thanks, that's what happens when I take a week long break from Go :)

File: pkg/quay/quay_test.go (hunk 2 above)
Comment (tnevrlka) [reply pkg/quay/quay_test.go]: Thank you. I rewrote the tests and I believe it solved the uncertainty about `.JSON(...)`

**Review by Michkov — APPROVED** [review]

**Pull request merged**